pkg/trainer/training.go

```diff
@@ -386,11 +399,11 @@ func (j *TrainingJob) Reconcile(config *tfv1alpha1.ControllerConfig, enableGangS
 		// TODO(jlewi): We should update the Phase if we detect the job is done.
 		if state == tfv1alpha1.StateFailed {
 			j.contextLogger.Errorf("Master failed Job: %v.", j.job.ObjectMeta.Name)
-			j.status.Phase = tfv1alpha1.TFJobPhaseCleanUp
+			j.status.Phase = tfv1alpha1.TFJobPhaseDone
```

Comment:
Why did you remove the CleanupPhase? Why don't you just call deleteResourcesByCleanPolicy in the CleanupPhase?

I don't think we want to do any work in the Done state because we would keep redoing that work even though there is no reason to.

Reply:
I just replace `CleanupPhase` with `DonePhase` when the state is `succeed` or `failed`. I think the `CleanupPhase` is set when `TFJob` is deleted， and it will clean all the resources belong to it. 

Referring https://github.com/kubeflow/tf-operator/pull/512/files#diff-ff02f3a407c605659e832c1863eab7b4L411

Reply:
I don't understand why you are changing the phases as opposed to changing the logic of the existing CleanupPhase.

We already have a CleanupPhase that is deleting all the resources. So instead of skipping the CleanupPhase and then calling deleteResourcesByCleanPolicy; just change the logic in the CleanUpPhase to call deleteResourcesByCleanPolicy.